daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1700,6 +1706,23 @@ func (devices *DeviceSet) Status() *Status {
 	return status
 }
 
+// Status returns the current status of this deviceset
+func (devices *DeviceSet) ExportDeviceMetadata(hash string) (*DeviceMetadata, error) {
+	info, err := devices.lookupDevice(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	info.lock.Lock()
```

Comment:
Right now nobody is changing the information I am fetching. But DeviceSize is something which can be changed in principle. We don't have an API for that yet but who knows we provide one in future.

So I would prefer to keep info.lock.Lock() around.